ui/src/cells/actions/thunks.ts

```diff
@@ -0,0 +1,162 @@
+// Libraries
+import {normalize} from 'normalizr'
+
+// APIs
+import * as api from 'src/client'
+import * as dashAPI from 'src/dashboards/apis'
+
+// Schemas
+import * as schemas from 'src/schemas'
+
+// Actions
+import {refreshDashboardVariableValues} from 'src/dashboards/actions/thunks'
+import {setView} from 'src/dashboards/actions/views'
+import {notify} from 'src/shared/actions/notifications'
+import {setCells, setCell, removeCell} from 'src/cells/actions/creators'
+
+// Utils
+import {getClonedDashboardCell} from 'src/dashboards/utils/cellGetters'
+
+// Constants
+import * as copy from 'src/shared/copy/notifications'
+
+// Types
+import {
+  Dashboard,
+  NewView,
+  Cell,
+  GetState,
+  RemoteDataState,
+  NewCell,
+  DashboardEntities,
+  ResourceType,
+  CellEntities,
+} from 'src/types'
+
+// Utils
+import {getViewsForDashboard} from 'src/dashboards/selectors'
+import {getNewDashboardCell} from 'src/dashboards/utils/cellGetters'
+import {getByID} from 'src/resources/selectors'
+
+export const deleteCell = (dashboardID: string, cellID: string) => async (
+  dispatch,
+  getState: GetState
+): Promise<void> => {
+  try {
+    const views = getViewsForDashboard(getState(), dashboardID).filter(
+      view => view.cellID !== cellID
+    )
+
+    await Promise.all([
+      api.deleteDashboardsCell({dashboardID: dashboardID, cellID: cellID}),
+      dispatch(refreshDashboardVariableValues(dashboardID, views)),
+    ])
+
+    dispatch(removeCell({dashboardID, id: cellID}))
+    dispatch(notify(copy.cellDeleted()))
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+export const createCellWithView = (
+  dashboardID: string,
+  view: NewView,
+  clonedCell?: Cell
+) => async (dispatch, getState: GetState): Promise<void> => {
+  const state = getState()
+
+  let dashboard = getByID<Dashboard>(
+    state,
+    ResourceType.Dashboards,
+    dashboardID
+  )
+
+  try {
+    if (!dashboard) {
+      const resp = await api.getDashboard({dashboardID})
+      if (resp.status !== 200) {
+        throw new Error(resp.data.message)
+      }
+
+      const {entities, result} = normalize<
+        Dashboard,
+        DashboardEntities,
+        string
+      >(resp.data, schemas.dashboard)
+
+      dashboard = entities.dashboards[result]
```

Comment:
this looks super familiar! saving the wrapping function (that encapsulates this logic and defines types) for a later PR?

Reply:
I think this is a great idea.  Something like `getResourceFromEntity(entities, ResourceType)`.  I'll def do that in the `views` pr